OrangeHRMTestFramework/PageObjects/OrangeHRM/DataGrids/BasicDataGrid.cs

```diff
@@ -0,0 +1,131 @@
+                isPaginationDisplayed = CheckIfPaginationButtonDisplayed();
+            }
+            WebDriverFactory.Driver.Navigate().Refresh();
+
+            return listOfColumnValuesFromAllPages;
+        }
+
+
+        public void ClickPaginationButton()
+        {
+            if (CheckIfPaginationButtonDisplayed())
+            {
+                var paginationButton = new OrangeWebElement(By.XPath(_paginationLocator));
+                paginationButton.ClickWithScroll();
+            }
+        }
+
+        public string GetValueByColumnNameAndRowIndex(string columnName, int rowNumber)
+        {
+            WaitUntillDataGridIsDisplayed();
+            var listOfColumnValues = GetCellValuesByColumnName(columnName);
+            var resultValue = listOfColumnValues[rowNumber - 1];
+
+            return resultValue;
+        }
+
+        public void ClickDeleteButtonByRowNumber(int rowNumber)
+        {
+            ClickActionButtonByRowIndex(1, rowNumber);
+        }
+
+        public void ClickEditButtonByRowNumber(int rowNumber)
+        {
+            ClickActionButtonByRowIndex(2, rowNumber);
+        }
+
+        public void SortDescByColumnName(string columnName)
+        {
+            var listOfColumnNamesWithSortOption = new List<string>();
+            var columnNamesWithSortingXPath = "//i[contains(concat('', @class, ''), 'oxd-table-header-sort-icon')]//ancestor::div[@role='columnheader']";
+            var sortIconXpath = "(//div[@role='columnheader']//i[contains(concat('', @class, ''), 'oxd-table-header-sort-icon')])[{0}]";
+            var sortDescDropdownOption = new OrangeWebElement(By.XPath("(//div[contains(concat('', @class, ''), '--active')]//li[@class='oxd-table-header-sort-dropdown-item']/span)[2]"));
+            var listOfColumnElementsWithSorting = _tableElement.FindElements(By.XPath(columnNamesWithSortingXPath));
+
+            foreach (var column in listOfColumnElementsWithSorting)
+            {
+                listOfColumnNamesWithSortOption.Add(column.Text);
+            }
+
+            var columnId = listOfColumnNamesWithSortOption.IndexOf(columnName) + 1;
+            var sortElement = new OrangeWebElement(By.XPath(string.Format(sortIconXpath, columnId)));
+            sortElement.ClickWithScroll();
+            sortDescDropdownOption.Click();
+            WaitUntillDataGridIsDisplayed();
+        }
+
+        private int GetColumnId(string columnName)
+        {
+            var columnsXpath = "//div[@role='columnheader']";
+            var allColumns = _tableElement.FindElements(By.XPath(columnsXpath));
+            var selectColumn = allColumns.Select(x => x).FirstOrDefault(x => x.Text == columnName);
+            var index = allColumns.IndexOf(selectColumn);
+
+            return index + 1;
+        }
+
+        private void WaitUntillDataGridIsDisplayed()
+        {
+            var rowElement = new OrangeWebElement(By.XPath("//div[@role='row']"));
+            if (!rowElement.Displayed)
```

Comment:
I think this condition is not required and you can leave only
rowElement.WaitUntilDisplayed();

Reply:
Fixed